lib/utils.js

```diff
@@ -37,36 +37,36 @@ function getCallTypeFromDescriptor (descriptor) {
 function getServiceDefintions (proto) {
   const services = {}
 
-  traverse(proto).forEach(function (v0) {
-    traverse(v0).forEach(function (v) {
-      if (isService(v)) {
-        const srviceObj = v.service || v
-        const vKeys = _.keys(srviceObj)
-        const name = getServiceNameFromPath(srviceObj[vKeys[0]].path)
-
-        if (services[name]) {
-          return
-        }
-
-        const shortServiceName = getShortServiceNameFromPath(srviceObj[vKeys[0]].path)
-
-        services[name] = {}
-        services[name].shortServiceName = shortServiceName
-        services[name].fullServiceName = name
-        services[name].service = srviceObj
-
-        const methods = {}
-        _.forOwn(srviceObj, (m, k) => {
-          methods[m.path] = _.pick(m, METHOD_PROPS)
-          methods[m.path].name = getMethodNameFromPath(m.path)
-          methods[m.path].fullName = m.path
-          methods[m.path].package = getPackageNameFromPath(m.path)
-          methods[m.path].service = shortServiceName
-        })
-
-        services[name].methods = methods
+  traverse(proto).forEach(function (v) {
+    if (isService(v)) {
+      const srviceObj = v.service || v
+      const vKeys = _.keys(srviceObj)
+      const name = getServiceNameFromPath(srviceObj[vKeys[0]].path)
+
+      if (services[name]) {
+        return
       }
-    })
+
+      const shortServiceName = getShortServiceNameFromPath(srviceObj[vKeys[0]].path)
+
+      services[name] = {}
+      services[name].shortServiceName = shortServiceName
+      services[name].fullServiceName = name
+      services[name].service = srviceObj
+
+      const methods = {}
+      _.forOwn(srviceObj, (m, k) => {
+        methods[m.path] = _.pick(m, METHOD_PROPS)
+        methods[m.path].name = getMethodNameFromPath(m.path)
+        methods[m.path].fullName = m.path
+        methods[m.path].package = getPackageNameFromPath(m.path)
+        methods[m.path].service = shortServiceName
+      })
+
+      services[name].methods = methods
+    } else if (v && v.type) {
+      this.update(v, true)
```

Comment:
Nice catch! Since we're not creating a new scope and using `traverse.update` method, I think adding a comment will improve the maintainability/readability a lot.